src/code_08026044.c

```diff
@@ -436,3 +436,583 @@ void sub_08026FAC(struct Cutscene *arg0) {
+        arg0->flags &= ~(6 << i * 3);
+    }
+    arg0->unk0 = sub_08028758;
+}
+
+void sub_08028808(struct Cutscene *arg0);
+void sub_08028758(struct Cutscene *arg0) {
+    if (arg0->flags & 2) {
+        u16 i;
+        for (i = 0; i < 4; ++i) {
+            SpriteInitNoTilesVramNoFunc(&arg0->unk4[i], 0x300, 0x49, 0, 0, 0xff, 0x10, i, arg0->unk2E4[i][0] >> 8, arg0->unk2E4[i][1] >> 8, 0x42000);
+            arg0->flags &= ~(2 << i * 3);
+            arg0->flags |= 4 << i * 3;
+        }
+        arg0->unk0 = sub_08028808;
+    }
+}
+
+void sub_080288D0(struct Cutscene *arg0);
+void sub_08028808(struct Cutscene *arg0) {
+    arg0->unk314 = 0xc800;
+    arg0->unk318 = -0x1400;
+    arg0->unk31C = 0x10000;
+    arg0->unk320 = 0xf000;
+    arg0->unk322 = 0x800;
+    arg0->unk324 = 0;
+    SpriteInitNoTilesVram(&arg0->unk144, 0x300, 0x2de, 0, 0, 0xff, 0x10, 0xe, arg0->unk314 >> 8, arg0->unk318 >> 8, 0x2000);
+    arg0->flags |= 0x400000;
+    m4aSongNumStart(SE_08D5F114);
+    arg0->unk362 = 0;
+    arg0->unk363 = 0;
+    arg0->unk35A = 0;
+    arg0->unk0 = sub_080288D0;
+}
+
+extern const s32 gUnk_082EB384[][3];
+void sub_08029E14(struct Cutscene *arg0);
+void sub_080288D0(struct Cutscene *arg0) {
+    u16 i;
+    s32 array[8][3];
+    u32 *b;
+    u8 eight = 8;
+    memcpy(array, gUnk_082EB384, sizeof(array));
+    arg0->unk320 += array[arg0->unk362][0];
+    arg0->unk322 += array[arg0->unk362][1];
+    arg0->unk314 += arg0->unk320;
+    arg0->unk318 += arg0->unk322;
+    arg0->unk144.x = arg0->unk314 >> 8;
+    arg0->unk144.y = arg0->unk318 >> 8;
+    for (i = 0; i < 4; ++i) {
+        b = &arg0->unk4[i].unk8;
+        if (arg0->unk314 < arg0->unk2E4[i][0]) {
+            *b |= 0x400;
+        }
+        else {
+            *b &= ~0x400;
+        }
+    }
+    if (++arg0->unk363 >= array[arg0->unk362][2]) {
+        arg0->unk363 = 0;
+        if (++arg0->unk362 >= eight) {
+            arg0->unk320 = 0;
+            arg0->unk322 = 0;
+            arg0->unk314 = (arg0->unk2E4[0][0] + arg0->unk2E4[1][0]) >> 1;
+            arg0->unk318 = 0x6c00;
+            arg0->unk0 = sub_08029E14;
+        }
+    }
+}
+
+void sub_08028AA8(struct Cutscene *arg0);
+void sub_08028A1C(struct Cutscene *arg0) {
+    u16 i;
+    for (i = 0; i < 4; ++i) {
+        struct Sprite *spr = &arg0->unk4[i];
+        SpriteInitNoTilesVramNoFunc(spr, 0x300, 0x49, 2, 0, 0xff, 0x10, i, arg0->unk2E4[i][0] >> 8, arg0->unk2E4[i][1] >> 8, spr->unk8);
+        arg0->flags &= ~(6 << i * 3);
+    }
+    arg0->unk0 = sub_08028AA8;
+}
+
+void sub_08029E2C(struct Cutscene *arg0);
+void sub_08028AA8(struct Cutscene *arg0) {
+    if (arg0->flags & 2) {
+        u16 i;
+        for (i = 0; i < 4; ++i) {
+            struct Sprite *spr = &arg0->unk4[i];
+            SpriteInitNoTilesVramNoFunc(spr, 0x300, 0, 0, 0, 0xff, 0x10, i, arg0->unk2E4[i][0] >> 8, arg0->unk2E4[i][1] >> 8, spr->unk8);
```

Comment:
Is this type of workaround ok? There are some instances where they don't write to unk8 and others, to x and y. Felt kind of weird to write a separate macro for those situations, considering they don't happen that often.

Reply:
It's fine